test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                await this.usdc
+                    .connect(this.deployer)
+                    .approve(this.subject.address, BigNumber.from(10001));
+                await expect(
+                    this.subject
+                        .connect(this.deployer)
+                        .deposit(
+                            [BigNumber.from(10001), BigNumber.from(10001)],
+                            BigNumber.from(1),
+                            []
+                        )
+                ).to.emit(this.subject, "Deposit");
+            });
+
+            describe("edge cases:", () => {
+                describe("when deposit is enabled", () => {
+                    it(`reverts with ${Exceptions.FORBIDDEN}`, async () => {
+                        await this.erc20RootVaultGovernance
+                            .connect(this.admin)
+                            .setOperatorParams({
+                                disableDeposit: true,
+                            });
+                        await expect(
+                            this.subject.deposit(
+                                [BigNumber.from(1), BigNumber.from(1)],
+                                BigNumber.from(1),
+                                []
+                            )
+                        ).to.be.revertedWith(Exceptions.FORBIDDEN);
+                    });
+                });
+
+                describe("when there is no depositor in allow list", () => {
+                    it(`reverts with ${Exceptions.FORBIDDEN}`, async () => {
+                        await expect(
+                            this.subject.deposit(
+                                [BigNumber.from(10001), BigNumber.from(10001)],
+                                BigNumber.from(randomInt(100)),
+                                []
+                            )
+                        ).to.be.revertedWith(Exceptions.FORBIDDEN);
+                    });
+                });
+
+                describe("when there is a private vault in delayedStrategyParams", () => {
+                    it(`reverts with ${Exceptions.FORBIDDEN}`, async () => {
+                        const params = {
+                            strategyTreasury: randomAddress(),
+                            strategyPerformanceTreasury: randomAddress(),
+                            privateVault: true,
+                            managementFee: BigNumber.from(randomInt(10 ** 6)),
+                            performanceFee: BigNumber.from(randomInt(10 ** 6)),
+                            depositCallbackAddress:
+                                ethers.constants.AddressZero,
+                            withdrawCallbackAddress:
+                                ethers.constants.AddressZero,
+                        };
+                        await this.erc20RootVaultGovernance
+                            .connect(this.admin)
+                            .stageDelayedStrategyParams(
```

Comment:
maybe I am wrong, but should this call be followed by commitDelayedStrategyParams after waiting for governanceDelay to pass
Otherwise delayedStrategyParams will not change